metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/config/ElasticConfig.java

```diff
@@ -0,0 +1,45 @@
+package eu.europeana.metis.authentication.rest.config;
+
+import static java.util.stream.Collectors.joining;
+
+import co.elastic.apm.attach.ElasticApmAttacher;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "elastic")
+@ConditionalOnProperty(value = "elastic.apm.enabled", havingValue = "true")
+public class ElasticConfig {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticConfig.class);
+  private static final String NEW_LINE_TAB = "\n\t";
+  private final Map<String, String> apmProperties = new HashMap<>();
+
+  /**
+   * The name of this method has to be as the key in the application.yml elastic: apm: ...
+   *
+   * @return map of elastic apm agent properties
+   */
+  public Map<String, String> getApm() {
+    return this.apmProperties;
+  }
+
+  @Override
+  public String toString() {
+    return NEW_LINE_TAB + apmProperties.entrySet().stream().map(Objects::toString)
```

Comment:
Why add the `NEW_LINE_TAB` in front of the string? Is it needed for some kind of setup?

Reply:
You are right!
I copied this from sandbox in fact but I didn't think about it too much.
This is used for the debugging log below. I've updated it to a better approach.